### PR TITLE
tests in JNC/tests can be run

### PR DESCRIPTION
**Pull request by martin-volf**
The directory `tests/` contains a set of unit tests for the jnc pyang plugin and several larger-scale tests for the whole package. They can be run now all or selectively using `pytest`.

Some of the unit tests and the large `12-test-augment` are failing; I'll open issue reports for those.

### CODE REVIEW

File: tests/11-test-notification/confd.conf.part [hunk 1]
```diff
@@ -0,0 +1,21 @@
+      <notification>
```
Comment (jomiklos) [inline tests/11-test-notification/confd.conf.part:1]: feels a bit hacky assuming some meta-nested tag start position, dropping/overloading all the contents that may change order in original confd.conf, leading to loss of some config/params...

File: tests/11-test-notification/src/app/Client.java [hunk 2]
```diff
@@ -136,16 +136,16 @@ public static void main(String[] args) throws IOException, JNCException {
 
         // Loop for 10000 iterations, print every 1000 received notification
         System.out.println("Waiting for more \"interface\" notifications...");
-        System.out.println("Only every 100 notification is printed.");
+        System.out.println("Only every 1000 notification is printed.");
```
Comment (jomiklos) [inline tests/11-test-notification/src/app/Client.java:139]: 1000th

File: tests/test_full_jnc.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    def cleanup(self):
+        self.jnc_runner.close()
+        confd = pexpect.spawn('make stop', **self.px_args)
+        confd.expect(pexpect.EOF)
+        confd.close()
+
+
+@pytest.fixture(scope='function',
+                params=['1-test-union',
```
Comment (jomiklos) [inline tests/test_full_jnc.py:30]: possibly make some automated ls/grep for directories with "test" (or similar) to save on this hidden constant maintenance?

Comment (martin-volf) [reply tests/test_full_jnc.py:30]: Is it worth it? I do not think it is likely we'll be adding more tests here; if we start doing that, maybe we shall rewrite this then.

**Review by jojohans — APPROVED** [review]

**Comment by martin-volf** [comment]
One more fix added, I realized that the test result depended on what had been run before it.

**Pull request closed without merging**